File: Languages/Kotlin/Day023.kt
// Languages - Kotlin - Day 23
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a recursive function that computes the Power of a number (x^n) without using built-in power functions.


fun power(base: Double, exponent: Int): Double {
    return when {
        exponent == 0 -> 1.0
        exponent < 0 -> 1.0 / power(base, -exponent)
        else -> base * power(base, exponent - 1)
    }
}

fun main() {
    println("Enter the base number:")
    val baseInput = readLine()?.trim()
    val base = baseInput?.toDoubleOrNull()
    if (base != null) {
        println("Enter the exponent:")
        val exponentInput = readLine()?.trim()
        val exponent = exponentInput?.toIntOrNull()
        if (exponent != null) {
            if (base == 0.0 && exponent == 0) {
                println("0^0 is undefined.")
            } else {
                val result = power(base, exponent)
                println("$base^$exponent = $result")
            }
        } else {
            println("Invalid input for exponent.")
        }
    } else {
        println("Invalid input for base.")
    }
}

File: Languages/JavaScript/Day023.js
// Languages - JavaScript - Day 23
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a recursive function that computes the Power of a number (x^n) without using built-in power functions.


// Recursive function to calculate the power of a number
function power(base, exponent) {
	if (exponent === 0) {
		return 1;
	} else if (exponent < 0) {
		return 1 / power(base, -exponent);
	} else {
		return base * power(base, exponent - 1);
	}
}

// Get the base and exponent from the user
const baseInput = prompt("Enter the base number:").trim();
const base = parseFloat(baseInput);

if (!isNaN(base)) {
	const exponentInput = prompt("Enter the exponent:").trim();
	const exponent = parseInt(exponentInput);

	if (!isNaN(exponent)) {
		if (base === 0 && exponent === 0) {
			console.log("0^0 is undefined.");
		} else {
			const result = power(base, exponent);
			console.log(`${base}^${exponent} = ${result}`);
		}
	} else {
		console.log("Invalid input for exponent.");
	}
} else {
	console.log("Invalid input for base.");
}
